torch/optim/lr_scheduler.py

```diff
@@ -667,18 +669,25 @@ class SequentialLR(LRScheduler):
     """
 
     def __init__(self, optimizer, schedulers, milestones, last_epoch=-1, verbose="deprecated"):
-        for scheduler_idx in range(len(schedulers)):
-            if schedulers[scheduler_idx].optimizer != optimizer:
+        if len(schedulers) < 1:
+            raise ValueError(f"{self.__class__.__name__} expects at least one scheduler, but got no scheduler.")
+
+        for scheduler_idx, scheduler in enumerate(schedulers):
+            if not hasattr(scheduler, 'optimizer'):
+                raise TypeError(
+                    f"{self.__class__.__name__} at index {scheduler_idx} should have `optimizer` as its attribute."
+                )
+            if isinstance(scheduler, ReduceLROnPlateau):
```

Comment:
no, this check is worse than before, because we do support ReduceLROnPlateau now! The other one would be better, but I’m still not sure why the check needs to exist at all.

Reply:
Hi @janeyx99 
friendly ping. Just to let you know that I'm happy to provide any additional information or clarification if needed, and really thanks for your time for reviewing this PR.

Sincerely.

Reply:
Ah yikes, sorry this fell under my stack of notifications--don't be shy about pinging if I am not responding--I don't intend to ignore these!

I see your intention is to treat ReduceLROnPlateau differently due to it needing kwargs in step (and thus being different from other LRSchedulers), yes? If so, why is the check in SequentialLR and not also in Chained below? Are there any cases where having a list of schedulers including ReduceLROnPlateau is fine but would be invalidated by this check? If there are no use cases where people can use Chained/Sequential with ReduceLROnPlateau, then this check is fine, though it may be good to specify that the reason why is due to ReduceLROnPlateau requiring kwargs in step to be specified.

Reply:
Thank you~
To answer your question, yes, it indeed is the reason why I did this check, and thanks for pointing this out, this check should be applied to both Chain and Sequential scheduler. 
Should I then change the error message to the following and add it to both Chain and Sequential?
```python
if isinstance(scheduler, ReduceLROnPlateau):
        raise ValueError(
            f"{self.__class__.__name__} does not support `ReduceLROnPlateau` scheduler as it requires additional kwargs to be specified when calling `step`, but got one at index {scheduler_idx} in the given schedulers sequence."
        )
```

Reply:
Yes

Reply:
Fixed.